src/chainstate/stacks/index/storage.rs

```diff
@@ -1317,9 +1325,7 @@ impl <T: MarfTrieId> TrieFileStorage <T> {
 
             debug!("Flush: identifier of {} is {}", flush_options, block_id);
 
-            if self.unconfirmed {
-                self.cur_block_id = Some(block_id);
-            }
+            self.cur_block_id = None;
```

Comment:
`cur_block_id` doesn't need to be reset here -- an update to #1763 fixes the root cause of this: `cur_block_id` should never be getting set for `last_extended_trie`, even if that's an unconfirmed block.